asdf/commands/tests/test_extract.py

```diff
@@ -1,11 +1,21 @@
 import os
+import sys
 
 import numpy as np
+import pytest
 from astropy.io.fits import HDUList, ImageHDU
 
 import asdf
 from asdf.commands import extract
-from asdf.fits_embed import AsdfInFits
+from asdf.exceptions import AsdfDeprecationWarning
+
+with pytest.warns(AsdfDeprecationWarning, match="AsdfInFits has been deprecated.*"):
+    # asdf.fits_embed is imported here
+    if "asdf.fits_embed" in sys.modules:
+        del sys.modules["asdf.fits_embed"]
+    import asdf.fits_embed
+    from asdf.fits_embed import AsdfInFits
```

Comment:
I added a comment to the code describing the need for this import.

In short, calling `del sys.modules['asdf.fits_embed']` isn't sufficient to allow for re-triggering the deprecation warning if an import like `from asdf.fits_embed import AsdfInFits` is used).

I chose adding the `import asdf.fits_embed` option to avoid changing the usage of AsdfInFits in test_extract.py but that should work as well (switching it to `asdf.fits_embed.AsdfInFits`). If you're prefer that alternative let me know.

Reply:
See https://github.com/asdf-format/asdf/pull/1337#discussion_r1110042655.